src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/configuration/TemplateInfoDaoImpl.java

```diff
@@ -0,0 +1,291 @@
+/* (c) 2021 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.featurestemplating.configuration;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.CatalogException;
+import org.geoserver.catalog.CatalogInfo;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geoserver.catalog.WorkspaceInfo;
+import org.geoserver.catalog.event.CatalogAddEvent;
+import org.geoserver.catalog.event.CatalogBeforeAddEvent;
+import org.geoserver.catalog.event.CatalogListener;
+import org.geoserver.catalog.event.CatalogModifyEvent;
+import org.geoserver.catalog.event.CatalogPostModifyEvent;
+import org.geoserver.catalog.event.CatalogRemoveEvent;
+import org.geoserver.config.GeoServerDataDirectory;
+import org.geoserver.platform.GeoServerExtensions;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.security.PropertyFileWatcher;
+
+/** A template info DAO that use a property file for persistence. */
+public class TemplateInfoDaoImpl implements TemplateInfoDao {
+
+    private TreeSet<TemplateInfo> templateDataSet;
+
+    private PropertyFileWatcher fileWatcher;
+
+    private GeoServerDataDirectory dd;
+
+    private Set<TemplateDAOListener> listeners;
+
+    private static final String PROPERTY_FILE_NAME = "features-templates-data.properties";
```

Comment:
Yes there are scenarios where template might be reused eg:

- include directives allows to reuse a template for multiple featureTypes.
- Since we have direct domain reference for Complex features the path is no more bounded to namespaces and in theory a same template file could work for different featureTypes if they come from the same source.

Regarding your questions:
- the previous behaviour is mantained: if no rule for content negotiation is defined on layer tab, then the plug-in will try to defined a template configured directly placed in featureType folder (eg. json-ld-template.json).
- the OpenSearch template mechanism should not be affected by this modifications since it use its own template loading mechanism, without using the TemplateLoader class that is the one handling rule evaluation.